packages/client/hmi-client/src/services/data.ts

```diff
@@ -562,47 +620,71 @@ const getDocumentById = async (docid: string) => {
 	return null;
 };
 
-const fetchData = async (term: string, searchParam?: SearchParameters) => {
+const fetchData = async (
+	term: string,
+	searchParam?: SearchParameters,
+	searchParamWithFacetFilters?: SearchParameters,
+	resultType?: string
+) => {
 	//
 	// call the different search sub-systems to retrieve results
 	// ideally, all such subsystems should be registered in an array, which will force refactoring of the following code
 	//
 
-	const promiseList = [] as Promise<SearchResults>[];
+	const promiseList = [] as Promise<FullSearchResults>[];
 
 	// xdd
-	const promise1 = new Promise<SearchResults>((resolve, reject) => {
-		try {
-			resolve(searchXDDArticles(term, searchParam?.xdd));
-		} catch (err: any) {
-			reject(new Error(`Error fetching XDD results: ${err}`));
-		}
-	});
-	promiseList.push(promise1);
+	if (resultType === ResourceType.XDD || resultType === ResourceType.ALL) {
+		// eslint-disable-next-line no-async-promise-executor
+		const promise1 = new Promise<FullSearchResults>(async (resolve, reject) => {
+			try {
+				const allData = await searchXDDArticles(term, searchParam?.xdd);
+				const allDataFilteredWithFacets = await searchXDDArticles(
+					term,
+					searchParamWithFacetFilters?.xdd
+				);
+				resolve({ allData, allDataFilteredWithFacets });
+			} catch (err: any) {
+				reject(new Error(`Error fetching XDD results: ${err}`));
+			}
+		});
+		promiseList.push(promise1);
+	}
 
 	// models
-	const promise2 = new Promise<SearchResults>((resolve, reject) => {
-		try {
-			resolve(getModels(term, searchParam?.model));
-		} catch (err: any) {
-			reject(new Error(`Error fetching model results: ${err}`));
-		}
-	});
-	promiseList.push(promise2);
+	if (resultType === ResourceType.MODEL || resultType === ResourceType.ALL) {
+		const promise2 = new Promise<FullSearchResults>((resolve, reject) => {
+			try {
+				resolve(getModels(term, searchParamWithFacetFilters?.model));
+			} catch (err: any) {
+				reject(new Error(`Error fetching model results: ${err}`));
+			}
+		});
+		promiseList.push(promise2);
+	}
 
 	// datasets
-	const promise3 = new Promise<SearchResults>((resolve, reject) => {
-		try {
-			resolve(getDatasets(term, searchParam?.dataset));
-		} catch (err: any) {
-			reject(new Error(`Error fetching dataset results: ${err}`));
-		}
-	});
-	promiseList.push(promise3);
+	if (resultType === ResourceType.DATASET || resultType === ResourceType.ALL) {
+		const promise3 = new Promise<FullSearchResults>((resolve, reject) => {
+			try {
+				resolve(getDatasets(term, searchParamWithFacetFilters?.dataset));
+			} catch (err: any) {
+				reject(new Error(`Error fetching dataset results: ${err}`));
+			}
+		});
+		promiseList.push(promise3);
+	}
 
```

Comment:
I think it may be worth looping through `ResourceType` instead of having similar if statements.
```suggestion
Object.entries(ResourceType).forEach(([key]) => {
		console.log(ResourceType[key])
		if (resultType === ResourceType[key] || resultType === ResourceType.ALL) {
			// eslint-disable-next-line no-async-promise-executor
			const promise = new Promise<FullSearchResults>(async (resolve, reject) => {
				try {
					switch (ResourceType[key]) {
						case ResourceType.XDD:
							const allData = await searchXDDArticles(term, searchParam?.xdd);
							const allDataFilteredWithFacets = await searchXDDArticles(
								term,
								searchParamWithFacetFilters?.xdd
							);
							resolve({ allData, allDataFilteredWithFacets });
							break;
						case ResourceType.MODEL:
							resolve(getModels(term, searchParamWithFacetFilters?.model));
							break;
						case ResourceType.DATASET:
							resolve(getDatasets(term, searchParamWithFacetFilters?.dataset));
							break;
					}
				} catch (err: any) {
					reject(new Error(`Error fetching ${ResourceType[key]} results: ${err}`));
				}
			});
			promiseList.push(promise);
		}
	});```

Reply:
Agreed, but I would leave this to another time once more focused code refactoring is performed!